libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/VanishingRouteLine.kt

```diff
@@ -55,11 +61,19 @@ internal class VanishingRouteLine {
      * Initializes this class with the active primary route.
      */
     fun initWithRoute(route: DirectionsRoute) {
-        primaryRoutePoints = parseRoutePoints(route)
-        primaryRouteLineGranularDistances = MapboxRouteLineUtils.calculateRouteGranularDistances(
-            primaryRoutePoints?.flatList
-                ?: emptyList()
-        )
+        jobControl.scope.launch {
```

Comment:
Shouldn't this block be guarded with a mutex? Otherwise, we are having the same risks of competing updates as we did in the async route line PR.

Reply:
It seemed like overkill since the read and write of the variables in question are on the main thread. However I can add a mutex if you feel strongly that it should be there.

Reply:
I still see the same risk, if we do:
1. `initWithRoute` with a long route
2. `initWithRoute` with a short route

The `short`update will finish first and be overwritten by the `long` update, making the feature inoperable.

Reply:
I got pulled in different directions so I didn't get as much time with this as I wanted.

Going with the mutex adds a side effect I don't like. The method MapboxRouteLineApi::updateTraveledRouteLine has to change to a callback since it would have to call VanishingRouteLine::getTraveledRouteLineExpression which would have to become a suspend function.

Instead I added cancellation to VanishingRouteLine::initWithRoute. I can see it working but it's extremely difficult to write a unit test for because the unit tests force everything onto the main thread.